### PR TITLE
Refactor init

### PR DESCRIPTION
**Pull request by ururruu**
Split the process() function to separate validate() model_validation() and preview() so we only call validation when it is needed.
Add init() functions to processors so it is proper OOP.
Move h5 code to h5 class from base (no one would inherit from this base).
Unify input types for all functions (No FileStorage checks)

### CODE REVIEW

File: file_process/csv.py [hunk 1]
```diff
@@ -5,67 +5,65 @@
...
-            elif all_targets == 'some':
-                are_targets_valid = not target_names or any(elem in dataset_vars for elem in target_names)
-            else:
-                are_targets_valid = True
+        all_targets = metadata.get('require_all_targets', True)
+        if all_targets == 'all':
+            are_targets_valid = not target_names or all(elem in dataset_vars for elem in target_names)
             if not are_targets_valid:
-                raise ModelFileValidationTargetsError
+                raise NotAllTargetsError
```
Comment (pengjiang-superbio) [inline file_process/csv.py:54]: If we can tell user what target is missing that will be helpful

File: file_process/csv.py [hunk 2]
```diff
@@ -5,67 +5,65 @@
...
+        elif all_targets == 'some':
+            are_targets_valid = not target_names or any(elem in dataset_vars for elem in target_names)
+            if not are_targets_valid:
+                raise NotSomeTargetsError
+
         are_variables_valid = all(elem in dataset_vars.difference(target_names)
                                   for elem in var_names.difference(target_names))
         if not are_variables_valid:
             raise ModelFileValidationVariablesError
```
Comment (pengjiang-superbio) [inline file_process/csv.py:63]: If we can tell user what variable failed, that will be helpful

File: file_process/csv.py [hunk 3]
```diff
@@ -5,67 +5,65 @@
...
-        obs_preview = df.head(min(10, df.shape[0]))
-        return var_names, obs_preview
+    def _get_delimiter(self, file: BytesIO):
+        reader = pd.read_csv(file, sep=None, iterator=True, nrows=10)
+        delimiter = reader._engine.data.dialect.delimiter  # pylint: disable=protected-access
+        file.seek(0)
+        return delimiter
 
-    def model_file_validation(self, df: pd.DataFrame, model_metadata_file: BytesIO, need_target: bool = True):
+    def get_obs(self):
```
Comment (pengjiang-superbio) [inline file_process/csv.py:29]: What is obs? Maybe we should use the whole words instead of abbreviation.

**Review by pengjiang-superbio — CHANGES_REQUESTED** [review]

File: file_process/csv.py (hunk 2 above)
Comment (ururruu) [reply file_process/csv.py:63]: that is a good point

File: file_process/csv.py (hunk 3 above)
Comment (ururruu) [reply file_process/csv.py]: Changed to observations

**Review by pengjiang-superbio — CHANGES_REQUESTED** [review]

File: file_process/csv.py [hunk 4]
```diff
@@ -5,67 +5,65 @@
...
 
-    def process(self, file, model_metadata_file: BytesIO = None, **kwargs) -> (List[str], None, pd.DataFrame):
-        df = self.read_file(file, **kwargs)
-        if model_metadata_file:
-            self.model_file_validation(df, model_metadata_file)
-        var_names, obs_preview = self.get_preview_data(df)
-        return var_names, None, obs_preview
+        dataset_diff = dataset_vars.difference(target_names)
+        var_names_diff = var_names.difference(target_names)
+        difference = var_names_diff - dataset_diff
```
[the remark this answers is not in the document]
Comment (ururruu) [reply file_process/csv.py:61]: 1. I guess it's what is needed. We can ask Mario on the standup
3. I don't think there's any

**Review by pengjiang-superbio — APPROVED** [review]

**Pull request merged**